import_system_symbols_from_ipsw.py

```diff
@@ -223,17 +223,54 @@ def extract_symbols_from_one_ipsw_archive(
     span = sentry_sdk.Hub.current.scope.span
     with span.start_child(op="task", description="Extract IPSW archive"):
         extract_zip_archive(ipsw_archive_path, extract_dir)
-    plist_path = os.path.join(extract_dir, "Restore.plist")
-    (system_restore_image_filename, os_version, build_number) = read_restore_plist(plist_path)
-
-    # Starting iOS 16.0, dyld caches are in a different image
-    if prefix == "ios" and version.parse(os_version) >= version.parse("16.0"):
-        plist_path = os.path.join(extract_dir, "BuildManifest.plist")
-        (system_restore_image_filename, os_version, build_number) = read_build_manifest_plist(
-            plist_path
-        )
 
+    os_version, build_number = read_system_version_plist(extract_dir)
+    logging.info(f"Found image for {os_version} ({build_number}) in {extract_dir}")
+    parsed_version = version.parse(os_version)
+
+    # Starting iOS 16.0 and macOS 13.0, dyld caches are in a different image
+    if (
+        prefix == "macos"
+        and parsed_version >= version.parse("13.0")
+        or prefix == "ios"
+        and parsed_version >= version.parse("16.0")
+    ):
+        system_restore_image_filename = read_build_manifest_plist(extract_dir)
```

Comment:
this here yields a single value, whereas the other branch yields an iterator. why exactly is that?

Reply:
The `Restore.plist` contains 2 possible restore images, one for a base system and one seemingly being the real system. We were extracting from the base system so far which is why our symbols were incomplete. I'm returning both images here and will loop through both to extract symbols.

As for the `BuildManifest.plist`, we only found the `Cryptex1,SystemOS` image being useful to access the `dyld` we want, so that's why it's returning only one value. Maybe in the future we'll need/want to support multiple images.